### ISSUE TITLE
[FEATURE] Remove `Folded` separator when not editing

### ISSUE THREAD

**Issue by leoravagnani** [UeharaYou/HiddenBar#6]
Hey @UeharaYou! I've been out the loop on this project for a while, just saw your comment on my issue on the original repo pointing to this fork.

I was wondering if it's possible to remove the `Folded` separator when out of edit mode so it doesn't take any space in normal usage. As it currently stands it acts like there's an icon there - albeit invisible.

![Screenshot 2023-11-07 at 02 40 50](https://github.com/UeharaYou/HiddenBar/assets/13058201/93df03d3-a0b8-451d-b1af-53797d0a7e52)

If it's impossible, no biggie, I'll just live with it ¯\_(ツ)_/¯›
Thanks for picking up this project!

**Comment by UeharaYou** [comment]
@leoravagnani Hi! Thanks for the issue. It is theoretically possible to remove the separator completely when showing hidden icons, but it's a lot harder to put it back to where it should be *(Technically, AppKit doesn't want apps to mess around with other apps' status bar item, not even letting apps to decide where their status bar item would be ＿|￣|○ )*. There will be a LOT of complexity and head-boggling introduced just for fixing this issue, so I'm sorry that you have to live with that for now🥺. 

ps. There are other ways to get more control to the menu bar that I'm testing with (which will address this issue and more), but it not stable enough for the release.

**Comment by leoravagnani** [comment]
hmm gotcha! well if the labour of tackling this (non-)issue isn't worth the effort it's really no big deal.

What about handling the width of the "icon" instead of removing it altogether? Not sure if it's feasible either as things seemingly changed since Big Sur... I get the impression that icons (or rather the root view that contains it) have a fixed minimum width, being able only to expand but not compress beyond that threshold. 

Admittedly I don't know much about macOS as I'm an iOS guy, just throwing ideas around haha

**Comment by leoravagnani** [comment]
Or what if the arrow itself becomes the separator? Of course I can't vouch for every user's preferences, but I'm assuming most people would put it where anything left of the arrow is hidden (just as Window's task bar groomed us decades ago).

**Comment by UeharaYou** [comment]
@leoravagnani Yes, using the main toggle as the primary separator is the ideal way of implementation. However it is also hard because macOS stops rendering status bar items once they exceed render range of the status bar (even just a little bit). HiddenBar serves the naive way where the main toggle serves as the control, and the separator "pushes" other icons out (which also pushes itself out of the range). The main toggle will not show on screen if it were the "pusher", so it is an unwilling compromise. _:(´ཀ`」 ∠):

ps. The gap is caused by macOS inserting paddings between icons, even if the icon itself is 0 in width. There's a way to adjust this padding though.

**Comment by cowboyen** [comment]
Thanks for the project. Instead of not hiding some elements when always hidden was enabled for the older version, its now actually hiding them:)

**Comment by derpyjoystick** [comment]
> There will be a LOT of complexity and head-boggling introduced just for fixing this issue, so I'm sorry that you have to live with that for now🥺.

I know this is old, but an effective workaround would be to reduce the white space (padding or margins) to the left of the chevron when it is activated and showing the menu items.

<img width="178" height="48" alt="Image" src="https://github.com/user-attachments/assets/55f26bf6-eee7-4679-9f3c-0c134b77e703" />

If you can trim most of the space between the red lines, it would pretty closely mimic what the OP was suggesting.

Not saying do it, just offering a workaround for doing it all "properly."
